### PR TITLE
feat(pypi): search sourceUrl in project_urls

### PR DESCRIPTION
**Pull request by trim21**
search sourceUrl in project_urls

```json
{
  "info": {
    "name": "flexget",
    "home_page": "https://flexget.com",
    "project_urls": {
      "Forum": "https://discuss.flexget.com",
      "Homepage": "https://flexget.com",
      "Issue Tracker": "https://github.com/Flexget/Flexget/issues",
      "Repository": "https://github.com/Flexget/Flexget"
    }
  }
}
```

### CODE REVIEW

File: lib/datasource/pypi/index.ts [hunk 1]
```diff
@@ -59,16 +59,33 @@ async function getDependency(
...
+
+    if (dep.info && !dependency.sourceUrl && dep.info.project_urls) {
+      for (const [name, projectUrl] of Object.entries(
+        dep.info.project_urls as { [key: string]: string }
+      )) {
+        const lower = name.toLowerCase();
+        if (
+          lower.startsWith('repo') ||
+          lower === 'code' ||
+          /^https?:\/\/github\.com\/[^\\/]+\/[^\\/]+$/.exec(projectUrl)
```
Comment (viceice) [inline lib/datasource/pypi/index.ts:81]: Please move the regex to constant in module scope.

File: lib/datasource/pypi/index.ts [hunk 2]
```diff
@@ -59,16 +59,33 @@ async function getDependency(
...
           'https://'
         );
-      } else {
-        dependency.homepage = dep.info.home_page;
       }
     }
+
+    if (dep.info && !dependency.sourceUrl && dep.info.project_urls) {
+      for (const [name, projectUrl] of Object.entries(
+        dep.info.project_urls as { [key: string]: string }
```
Comment (viceice) [inline lib/datasource/pypi/index.ts:75]:
```suggestion
        dep.info.project_urls as Record<string, string>
```

File: lib/datasource/pypi/index.ts [hunk 3]
```diff
@@ -59,16 +59,33 @@ async function getDependency(
...
         dependency.sourceUrl = dep.info.home_page.replace(
           'http://',
           'https://'
         );
-      } else {
-        dependency.homepage = dep.info.home_page;
       }
     }
+
+    if (dep.info && !dependency.sourceUrl && dep.info.project_urls) {
```
Comment (viceice) [inline lib/datasource/pypi/index.ts:73]:
```suggestion
    if (dep.info?.project_urls && !dependency.sourceUrl) {
```

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by viceice — DISMISSED** [review]

**Comment by Trim21** [comment]
add pypi json type

File: lib/datasource/pypi/index.ts [hunk 4]
```diff
@@ -59,16 +72,31 @@ async function getDependency(
       );
       return null;
     }
+
     if (dep.info && dep.info.home_page) {
```
Comment (viceice) [inline lib/datasource/pypi/index.ts:76]:
```suggestion
    if (dep.info?.home_page) {
```

**Review by viceice — COMMENTED** [review]
just a small optimization

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**